src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/OffsetOfTests.cs

```diff
@@ -0,0 +1,364 @@
+            Assert.Equal(new IntPtr(8), Marshal.OffsetOf(t, "m_int1"));
+            Assert.Equal(new IntPtr(12), Marshal.OffsetOf(t, "m_byte2"));
+            Assert.Equal(new IntPtr(16), Marshal.OffsetOf(t, "m_int2"));
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || (RuntimeInformation.ProcessArchitecture != Architecture.X86))
+            {
+                Assert.Equal(new IntPtr(24), Marshal.OffsetOf(t, "m_double1"));
+                Assert.Equal(new IntPtr(32), Marshal.OffsetOf(t, "m_char1"));
+                Assert.Equal(new IntPtr(33), Marshal.OffsetOf(t, "m_char2"));
+                Assert.Equal(new IntPtr(34), Marshal.OffsetOf(t, "m_char3"));
+                Assert.Equal(new IntPtr(40), Marshal.OffsetOf(t, "m_double2"));
+                Assert.Equal(new IntPtr(48), Marshal.OffsetOf(t, "m_byte3"));
+                Assert.Equal(new IntPtr(49), Marshal.OffsetOf(t, "m_byte4"));
+                Assert.Equal(new IntPtr(56), Marshal.OffsetOf(t, "m_decimal1"));
+                Assert.Equal(new IntPtr(72), Marshal.OffsetOf(t, "m_char4"));
+            }
+            else
+            {
+                Assert.Equal(new IntPtr(20), Marshal.OffsetOf(t, "m_double1"));
+                Assert.Equal(new IntPtr(28), Marshal.OffsetOf(t, "m_char1"));
+                Assert.Equal(new IntPtr(29), Marshal.OffsetOf(t, "m_char2"));
+                Assert.Equal(new IntPtr(30), Marshal.OffsetOf(t, "m_char3"));
+                Assert.Equal(new IntPtr(32), Marshal.OffsetOf(t, "m_double2"));
+                Assert.Equal(new IntPtr(40), Marshal.OffsetOf(t, "m_byte3"));
+                Assert.Equal(new IntPtr(41), Marshal.OffsetOf(t, "m_byte4"));
+                Assert.Equal(new IntPtr(48), Marshal.OffsetOf(t, "m_decimal1"));
+                Assert.Equal(new IntPtr(64), Marshal.OffsetOf(t, "m_char4"));
+            }
+        }
+
+        [Fact]
+        public void TestFieldAlignment_Decimal()
+        {
+            var t = typeof(FieldAlignmentTest_Decimal);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || (RuntimeInformation.ProcessArchitecture != Architecture.X86))
+            {
+                Assert.Equal(96, Marshal.SizeOf(t));
+            }
+            else
+            {
+                Assert.Equal(88, Marshal.SizeOf(t));
+            }
+
+            Assert.Equal(new IntPtr(0), Marshal.OffsetOf(t, "b"));
+            Assert.Equal(new IntPtr(8), Marshal.OffsetOf(t, "p"));
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || (RuntimeInformation.ProcessArchitecture != Architecture.X86))
+            {
+                Assert.Equal(new IntPtr(88), Marshal.OffsetOf(t, "s"));
+            }
+            else
+            {
+                Assert.Equal(new IntPtr(80), Marshal.OffsetOf(t, "s"));
+            }
+        }
+
+        [Fact]
+        public void TestFieldAlignment_Guid()
+        {
+            var t = typeof(FieldAlignmentTest_Guid);
```

Comment:
Nit: var => Type (for this test and the others following the same pattern)

Reply:
Fixed, see in next commit